module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,257 @@
+
+            // TODO(cmurtagh) work out if gain is velocity or force
+            MotorVelocity velocity_msg;
+            velocity_msg.name     = command.id;
+            velocity_msg.velocity = command.gain;
+            to_send.motor_velocities.push_back(velocity_msg);
+
+            MotorTorque torque_msg;
+            torque_msg.name   = command.id;
+            torque_msg.torque = command.torque;
+            to_send.motor_torques.push_back(torque_msg);
+
+            // MotorPID ? Do we need to send this?
+        }
+    });
+}
+
+void Webots::send_player_details(const int& fd, const GlobalConfig& player_details) {
+    // TODO(cameron) resend if failes
+    std::vector<char> data = NUClear::util::serialise::Serialise<GlobalConfig>::serialise(player_details);
+    uint32_t N             = htonl(data.size());
+    send(fd, &N, sizeof(N), 0);
+    send(fd, data.data(), N, 0);
+}
+
+void Webots::translate_and_emit_sensor(const SensorMeasurements& sensor_measurements) {
+            // Read each field of msg, translate it to our protobuf and emit the data
+            auto sensor_data = std::make_unique<DarwinSensors>();
+
+
+            sensor_data->timestamp = sensor_measurements.time;  // Not sure if we want this or the timestamp on the received message.
+
+            // Vecotor3 is a neutron of 3 doubles
+
+            for (auto& position : sensor_measurements.position_sensors) {
+                // string name
+                // double valvue
+            }
+
+            for (auto& accelerometer : sensor_measurements.accelerometers) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& gyro : sensor_measurements.gyros) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& bumper : sensor_measurements.bumpers) {
+                // string name
+                // bool value
+            }
+
+            for (auto& force_3d : sensor_measurements.force3ds) {
+                // string name
+                // Vector3 value
+            }
+
+            for (auto& force_6d : sensor_measurements.force6ds) {
+                // string name
+                // Vector3 value
+            }
+
+            emit(sensor_data);
+
+            for (auto& camera : sensor_measurements.cameras) {
+                // Convert the incoming image so we can emit it to the PowerPlant.
+                auto compressed_image            = std::make_unique<CompressedImage>();
+                compressed_image->name           = camera.name;
+                compressed_image->dimensions.x() = camera.width;
+                compressed_image->dimensions.y() = camera.height;
+                compressed_image->format         = fourcc("JPEG");
+                compressed_image->data           = camera.image;
+                emit(compressed_image);
+            }
+
+            // Parse the messages from Webots and log them. Maybe check for certain things.
```

Comment:
This comment doesn't quite match what the switch is doing, since it's only logging errors and warnings. The comment implies to me that all messages are being logged.